File: DSA College/Tree/pre_order.c
#include<stdio.h>
#include<stdlib.h>

typedef struct node{
    int data;
    struct node *left;    
    struct node *right;    
}node;

node *createnode(int data){
    node *new=malloc(sizeof(node));
    new->data=data;
    new->left=NULL;
    new->right=NULL;

    return new;
}

//Pre - order Traversal

void Pre_Order(node *root){                         // MAIN CODE OF PRE ORDER
    node *ptr=root;
    if(ptr!=NULL){
        printf("%d ",ptr->data);

        Pre_Order(root->left);        
        Pre_Order(root->right);        
    }
}

int main(){

    //Node Creation

    node *root=createnode(1);
    node *p2=createnode(2);
    node *p3=createnode(3);
    node *p4=createnode(4);
    node *p5=createnode(5);
    node *p6=createnode(6);


    // Linking of Nodes

    root->left=p2;
    p2->left=p4;
    p2->right=p5;

    root->right=p3;
    p3->right=p6;

    /* Tree Looks Like
            1
          /  \
        2     3
       /  \     \
     4     5      6


    */

   printf("Pre-Order Traversal : ");
   Pre_Order(root);

    return 0;
}

File: DSA using Java/Tree/pre_order.java
import java.util.*;

public class pre_order{

    static class node{
        int data;
        node left;
        node right;

        node(int data){
            this.data=data;
            this.left=null;
            this.right=null;
        }

    }


    static class BinaryTree{
        static int idx=0;
        public static node buildTree(int nodes[]){
            idx++;
            if(nodes[idx]==-1){
                return null;
            }

            node newNode=new node(nodes[idx]);
            newNode.left=buildTree(nodes);
            newNode.right=buildTree(nodes);

            return newNode;

        }
    }

    public static void main(String[] args) {
        int nodes[]={1,2,4-1,-1,5,-1,-1,3,-1,6,-1,-1};   // pre order

        BinaryTree tree=new BinaryTree();
        node root=tree.buildTree(nodes);
        System.out.println(root);
        
    }
}